id/src/main/java/com/amplitude/id/utilities/PropertiesFile.kt

```diff
@@ -32,8 +32,12 @@ class PropertiesFile(directory: File, key: String, prefix: String, logger: Logge
     }
 
     private fun save() {
-        FileOutputStream(propertiesFile).use {
-            underlyingProperties.store(it, null)
+        try {
+            FileOutputStream(propertiesFile).use {
+                underlyingProperties.store(it, null)
+            }
+        } catch (e: IndexOutOfBoundsException) {
```

Comment:
Is this the only IO error that can happen? We might want to catch all IO errors

Reply:
Changed to more general exception catch